### PR TITLE
Use FileSystemStaticHandler for all webjars

### PR DESCRIPTION
**Pull request by Postremus**
The changes in this PR prevent copying all static resources needed for webjars (smallrye-health, swaggerui, graphql), and serves them directly from the jar instead.

I also replaced the extension specific handlers (e.g. SwaggerUiNotFoundHandler, SwaggerUiStaticHandler) with general purpose implementation (WebJarNotFoundHandler, WebJarStaticHandler) in vertx-http.

Same idea as in #22525.

This saves about 30ms of dev mode startup time.
Relates to #21552

### CODE REVIEW

File: extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiProcessor.java [hunk 1]
```diff
@@ -136,6 +136,7 @@ public void getSwaggerUiFinalDestination(
             byte[] indexHtmlContent = generateIndexHtml(openApiPath, swaggerUiPath, swaggerUiConfig, indexRootPathBuildItem);
             webJarBuildProducer.produce(
                     WebJarBuildItem.builder().artifactKey(SWAGGER_UI_WEBJAR_ARTIFACT_KEY) //
+                            .onlyCopyNonArtifactFiles(true)
```
Comment (gsmet) [inline extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiProcessor.java:139]: Why this change? Shouldn't it be the default if we generalize it?

**Review by gsmet — COMMENTED** [review]
Looks good to me and a nice cleanup.

I just added a small open question.

File: extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiProcessor.java (hunk 1 above)
Comment (gsmet) [reply extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiProcessor.java:139]: BTW, it's an open question, you're the one with the most knowledge of this stuff so I'm curious to hear your opinion.

Comment (Postremus) [reply extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiProcessor.java:139]: Setting `onlyCopyNonArtifactFiles` to default as `true`, could break quarkiverse members that have already updated to the new webjar infrastructure with 2.7.0.CR1.

I (personally) don't believe a .Final should intentionally break extensions.
WDYT?

**Comment by Postremus** [comment]
I fixed the vert.x 4.2.4 deprecations in WebJarStaticHandler, and disallowed rootfilesystemaccess.
This handler only needs to serve from META-INF anyway.

/cc @cescoffier

File: extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiProcessor.java (hunk 1 above)
Comment (gsmet) [reply extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiProcessor.java:140]: Well, if making it the default makes sense, I prefer we do it now rather than after because it will be far more of a pain to change it later.

Now, I don't know if it makes sense to make it the default - I will let you decide what makes more sense on this - but, if it does, let's do it now.

Comment (Postremus) [reply extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiProcessor.java]: Yep, defaulting to true makes sense, and is easy enough to change if one needs to.
Done.

**Comment by Postremus** [comment]
Test failures in CustomQuarkusDevModeConfigurationTest do not look related

**Review by gsmet — APPROVED** [review]

**Review by cescoffier — APPROVED** [review]

**Comment by gsmet** [comment]
I'm testing this locally and will merge if things are OK.

**Comment by gsmet** [comment]
Thanks @Postremus , very nice work!

File: extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLRecorder.java [hunk 2]
```diff
@@ -52,12 +56,16 @@
...
     public Handler<RoutingContext> uiHandler(String graphqlUiFinalDestination,
-            String graphqlUiPath, SmallRyeGraphQLRuntimeConfig runtimeConfig) {
+            String graphqlUiPath, List<FileSystemStaticHandler.StaticWebRootConfiguration> webRootConfigurations,
+            SmallRyeGraphQLRuntimeConfig runtimeConfig, ShutdownContext shutdownContext) {
 
         if (runtimeConfig.enable) {
-            return new SmallRyeGraphQLStaticHandler(graphqlUiFinalDestination, graphqlUiPath);
+            WebJarStaticHandler handler = new WebJarStaticHandler(graphqlUiFinalDestination, graphqlUiPath,
+                    webRootConfigurations);
+            shutdownContext.addShutdownTask(new ShutdownContext.CloseRunnable(handler));
```
Comment (gsmet) [inline extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLRecorder.java:65]: @mkouba @cescoffier I was going back to this patch and was wondering why we needed to do that? Shouldn't the handler be closed by Vert.x itself?

Comment (Postremus) [reply extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLRecorder.java:65]: https://github.com/quarkusio/quarkus/blob/main/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/devmode/FileSystemStaticHandler.java#L166

This is the handler in question which should be closed.
At first, I also did hopped that closeable are automaticly closed. I set a breakpoint in the close method though. Never got invoked.

This is why I decided to add the shutdownhandlers.

Is there any better / automatic way with Vert.x?

Comment (gsmet) [reply extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLRecorder.java:65]: Ah OK, I missed you specifically implemented `Closeable`. Maybe @mkouba will have some ideas about how we could improve it and close `Closeable` handlers automatically.
If not, that's not a biggy.

I was going through this patch again to move Quarkus GitHub App to this new pattern.
